python/python/bystro/supervised_ppca/gf_generative_pt.py

```diff
@@ -39,21 +39,26 @@
 None
 """
 import numpy as np
+from numpy.typing import NDArray
+from typing import Dict, List, Optional
```

Comment:
you can remove this line. Instead of using Dict, List, and Optional types, use:

`dict`, `list`, and `T | None`